packages/core/src/RoutingNode.ts

```diff
@@ -77,27 +79,26 @@ export class RoutingNode {
 		runIndex: number,
 		nodeType: INodeType,
 		executeData: IExecuteData,
-		nodeExecuteFunctions: INodeExecuteFunctions,
 		credentialsDecrypted?: ICredentialsDecrypted,
 		abortSignal?: AbortSignal,
 	): Promise<INodeExecutionData[][] | null | undefined> {
 		const items = inputData.main[0] as INodeExecutionData[];
 		const returnData: INodeExecutionData[] = [];
 
 		const closeFunctions: CloseFunction[] = [];
-		const executeFunctions = nodeExecuteFunctions.getExecuteFunctions(
+		const executeFunctions = new ExecuteContext(
 			this.workflow,
+			this.node,
+			this.additionalData,
+			this.mode,
 			this.runExecutionData,
 			runIndex,
 			this.connectionInputData,
 			inputData,
-			this.node,
-			this.additionalData,
 			executeData,
-			this.mode,
 			closeFunctions,
 			abortSignal,
-		);
+		) as IExecuteFunctions;
```

Comment:
This can be removed. It then shows that there is a type mismatch between the interface and the implementation:

https://github.com/n8n-io/n8n/blob/2855a89ab40378d8c4115e825950afc1b1541043/packages/workflow/src/Interfaces.ts#L857-L860

Second arg is optional.

https://github.com/n8n-io/n8n/blob/2855a89ab40378d8c4115e825950afc1b1541043/packages/core/src/node-execution-context/base-execute-context.ts#L94-L104

Second arg is required. Making it optional fixes the type error.